src/uu/cp/src/platform/linux.rs

```diff
@@ -193,3 +401,196 @@ pub(crate) fn copy_on_write(
     result.context(context)?;
     Ok(copy_debug)
 }
+
+fn handle_reflink_auto_sparse_always(
```

Comment:
please add rustdoc comment + unit test of this function

(same for the other functions)

Reply:
Sure, i will add documentation for the functions and by unit tests , testing the debug results for different types of files should be good enough right ?